Sources/SwiftDocC/Model/MarkdownOutput/Translation/MarkdownOutputMarkdownWalker.swift

```diff
@@ -0,0 +1,381 @@
+                // Do nothing
+            } else {
+                visit(Heading(level: 2, Text(heading)))
+            }
+        }
+        
+        section.content.forEach {
+            self.visit($0)
+        }
+    }
+        
+    mutating func startNewParagraphIfRequired() {
+        if !markdown.isEmpty, !markdown.hasSuffix("\n\n") { markdown.append("\n\n") }
+    }
+}
+
+extension MarkdownOutputMarkupWalker {
+    
+    mutating func defaultVisit(_ markup: any Markup) -> () {
+        var output = markup.format()
+        if let indentationToRemove, output.hasPrefix(indentationToRemove) {
+            output.removeFirst(indentationToRemove.count)
+        }
+        markdown.append(output)
+    }
+        
+    mutating func visitHeading(_ heading: Heading) -> () {
+        startNewParagraphIfRequired()
+        markdown.append(heading.detachedFromParent.format())
+        if heading.level > 1 {
+            lastHeading = heading.plainText
+        }
+    }
+    
+    mutating func visitUnorderedList(_ unorderedList: UnorderedList) -> () {
+        guard isRenderingLinkList else {
+            return defaultVisit(unorderedList)
+        }
+        
+        startNewParagraphIfRequired()
+        for item in unorderedList.listItems {
+            item.children.forEach { visit($0) }
+            startNewParagraphIfRequired()
+        }
+    }
+    
+    mutating func visitImage(_ image: Image) -> () {
+        guard let source = image.source else {
+            return
+        }
+        let unescaped = source.removingPercentEncoding ?? source
+        var filename = source
+        if
+            let resolved = context.resolveAsset(named: unescaped, in: identifier, withType: .image), let first = resolved.variants.first?.value {
+            filename = first.lastPathComponent
+        }
+                    
+        markdown.append("![\(image.altText ?? "")](images/\(context.inputs.id)/\(filename))")
+    }
+       
+    mutating func visitCodeBlock(_ codeBlock: CodeBlock) -> () {
+        startNewParagraphIfRequired()
+        markdown.append(codeBlock.detachedFromParent.format())
+    }
+    
+    mutating func visitSymbolLink(_ symbolLink: SymbolLink) -> () {
+        guard
+            let destination = symbolLink.destination,
+            let resolved = context.referenceIndex[destination],
+            let node = context.topicGraph.nodeWithReference(resolved)
+        else {
+            return defaultVisit(symbolLink)
+        }
+        
+        let linkTitle: String
+        var linkListAbstract: (any Markup)?
+        if
+            isRenderingLinkList,
+            let doc = try? context.entity(with: resolved),
+            let symbol = doc.semantic as? Symbol
+        {
+            linkListAbstract = (doc.semantic as? Symbol)?.abstract
+            if let fragments = symbol.navigator {
+                linkTitle = fragments
+                    .map { $0.spelling }
+                    .joined(separator: " ")
```

Comment:
Question: why is this adding a space between the fragments? The navigator fragments should already contains `.text` tokens with whitespace.